src/community/schemaless-features/schemaless-core/src/main/java/org/geoserver/schemalessfeatures/data/SchemalessDataAccess.java

```diff
@@ -0,0 +1,72 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.schemalessfeatures.data;
+
+import java.io.IOException;
+import java.util.List;
+import org.geotools.data.DataAccess;
+import org.geotools.data.ServiceInfo;
+import org.geotools.feature.NameImpl;
+import org.geotools.filter.FilterCapabilities;
+import org.opengis.feature.Feature;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.Name;
+
+public abstract class SchemalessDataAccess implements DataAccess<FeatureType, Feature> {
```

Comment:
Not asking to change this set of classes, but looking at them, I don't see much that has to do with schemaless anyways... they look more like a first cut at a ContentDataStore family for complex features. Something to remember in the future.